src/api.js

```diff
@@ -1373,6 +1373,8 @@ api.declare({
     this.workerInfo.seen(provisionerId, workerType, workerGroup, workerId),
   ]);
 
+  await this.workerInfo.taskSeen(provisionerId, workerType, workerGroup, workerId, result);
```

Comment:
This should get done in the same Promise.all above, so that it occurs in parallel.

Reply:
Although that leaves a bit of a race between workerInfo.seen creating a worker entity and workerInfo.taskSeen updating its list of tasks.  Could this all be included in workerInfo.seen, by passing `tasks` to it?

Reply:
@djmitche Hm, tasks (`result`) is only available after `this.workClaimer.claim` resolves and so if this happens in parallel to workerInfo.seen, I won't be able to pass `tasks` to it.

Reply:
Oh, I didn't see that.  OK, that solves the race, too.